### PR TITLE
Data Processing Refactor

### PR DESCRIPTION
**Pull request by DnOberon**
## Description

- Metatype Relationship Pair's `Filter` type now returns a composite type with the origin and destination metatype names, as well as the relationship pair name.
- Filters for Metatype Relationship Pairs
- Corrected problem with data import processing where edges couldn't see previously inserted nodes, now uses the same transaction for node and edge insertion for each data import. See the data processing loop for more information.
- Data import records no longer have `errors, stopped_at, started_at` fields. Instead we rely on `modified_at` and two new fields - `status` and `status_message` to determine disposition of import.
- Edges now store correctly when Deep Lynx is provided with a potential edge with the same existing original_data_id as an existing edge
- Removed a bunch of the Readmes and converted them into being part of the wiki

## Motivation and Context
Various changes needed for the type mapping system in the admin GUI.

## How Has This Been Tested?
Tests have been run and pass. I've also used Postman to verify new endpoints

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/data_importing/httpImpl.ts [hunk 1]
```diff
@@ -196,11 +196,11 @@ export class HttpImpl implements DataSource {
             const lastImport = await ImportStorage.Instance.RetrieveLast(this.dataSourceT.id!);
 
             let lastImportTime = "";
-            if(!lastImport.isError && lastImport.value.stopped_at != null) {
-                lastImportTime = lastImport.value.stopped_at.toUTCString()
+            if(!lastImport.isError && lastImport.value.status === "completed") {
```
Comment (jerenmb) [inline src/data_importing/httpImpl.ts:199]: Is this project making use of enums? If so, would this be a good candidate? If not, has that design decision been made explicit in documentation?

Comment (DnOberon) [reply src/data_importing/httpImpl.ts:199]: unfortunately I don't think I can use an enum type with `io-ts`, that's why I've stuck with string literals.

File: src/data_processing/processing.ts [hunk 2]
```diff
@@ -111,43 +114,61 @@ export class DataSourceProcessor {
...
-        const processQueries: QueryConfig[] = []
+        // we must wrap this insert as a transaction so that if there are errors
+        // we are capable of rolling back Deep Lynx's data to a point before the
+        // import. Because we need the transaction client for edge verification
+        // we'll use the transaction primitives of a storage class.
+        const transaction = await GraphStorage.Instance.startTransaction()
 
         if(nodesToInsert.length > 0) {
             // insert all nodes first, then edges
             const insertedNodes = await NodeStorage.Instance.CreateOrUpdateStatement(this.dataSource.container_id!, this.graphID, nodesToInsert)
```
[the remark this answers is not in the document]
Comment (DnOberon) [reply src/data_processing/processing.ts:127]: That's correct. I can make a change to the error messages

**Review by jerenmb — APPROVED** [review]

File: src/data_processing/processing.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (DnOberon) [reply src/data_processing/processing.ts]: Individual data points can now have a unique error message, helping us pinpoint on which point of data an import failed on.

**Pull request merged**